server/src/main/java/io/crate/planner/operators/HashJoin.java

```diff
@@ -237,8 +239,8 @@ public LogicalPlan replaceSources(List<LogicalPlan> sources) {
 
     @Override
     public LogicalPlan pruneOutputsExcept(TableStats tableStats, Collection<Symbol> outputsToKeep) {
-        ArrayList<Symbol> lhsToKeep = new ArrayList<>();
-        ArrayList<Symbol> rhsToKeep = new ArrayList<>();
+        HashSet<Symbol> lhsToKeep = new HashSet<>();
```

Comment:
Not sure if we should use a `LinkedHashSet` here instead. Operators like `eval` use `outputsToKeep` to create their `outputs`, so by using `HashSet` we could get flaky tests